client/src/lib/constants.ts

```diff
@@ -1,3 +1,27 @@
 const ADDRESS_REGEXP = /^0x[0-9a-fA-F]{40}$/;
```

Comment:
Since we're using [ethers](https://docs.ethers.io/ethers.js/html/api-utils.html#hex-strings) library I check that they have a `isHexString` util function that looks like:

```typescript
export function isHexString(value: any, length?: number): boolean {
    if (typeof(value) !== 'string' || !value.match(/^0x[0-9A-Fa-f]*$/)) {
        return false
    }
    if (length && value.length !== 2 + 2 * length) { return false; }
    return true;
}
```

I think we can safely remove this `RegExp` and use `ethers.isHexString` alongside with yup [test](https://github.com/jquense/yup#mixedtestname-string-message-string--function-test-function-schema) method.

We're using this RegExp on `client/src/lib/schemas.ts` in line 48

Old code:
```typescript
const PoapEventSchema = yup.object().shape({
  // ... many other validations
  signer: yup
    .string()
    .matches(ADDRESS_REGEXP, 'Must be a valid Ethereum Address')
    .nullable(),
});
```

New code:
```typescript
signer: yup
    .string()
    .test(
      'is-signer-an-address',
      '${path} is not a valid Ethereum Address',
      signer => ethers.isHexString(signer),
    .nullable(),
```

Reply:
It's important that we validate the address length, so I added to your example the `length` param to the `isHexString` function.